lib/autoproj/daemon/github_watcher.rb

```diff
@@ -1,189 +1,224 @@
-            # @param [String] name The name of the repository
-            # @param [String] branch The branch to watch
-            # @return [nil]
-            def add_repository(owner, name, branch = 'master')
-                @start_time ||= login
-                watched_repositories <<
-                    Repository.new(owner, name, branch, @start_time)
-                nil
+            # @param [String] owner
+            # @param [String] name
+            # @param [String] branch
+            # @return [Boolean]
+            def to_mainline?(owner, name, branch)
+                packages.any? do |pkg|
+                    pkg.owner == owner && pkg.name == name && pkg.branch == branch
+                end
             end
 
-            # Adds a block that will be called whenever a pull request is opened
-            # in any of the watched repositories.
-            #
-            # @yield [repo, options] Gives the event description to the block
-            # @yieldparam [String] repo The repository name in the format user/repo_name
-            # @yieldparam [Hash] options A hash with pull request meta data
-            # @return [nil]
-            def add_pullrequest_hook(&hook)
-                @pullrequest_hooks << hook
-                nil
+            # @param [String] owner
+            # @param [String] name
+            # @param [String] branch
+            # @return [Boolean]
+            def to_pull_request?(owner, name, branch)
+                cache.pull_requests.any? do |pr|
+                    pr.head_owner == owner && pr.head_name == name &&
+                        pr.head_branch == branch
+                end
             end
 
-            # Adds a block that will be called whenever a push to a watched branch
-            # or to a branch that has a PR open to a watched branch is detected.
-            #
-            # @yield [repo, options] Gives the event description to the block
-            # @yieldparam [String] repo The repository name in the format user/repo_name
-            # @yieldparam [Hash] options A hash with push meta data
-            # @return [nil]
-            def add_push_hook(&hook)
-                @push_hooks << hook
-                nil
+            # @param [Array<Github::PushEvent, Github::PullRequestEvent>] events
+            # @return [Array<Github::PushEvent, Github::PullRequestEvent>]
+            def filter_events(events)
+                events.select do |event|
+                    if event.kind_of? Github::PushEvent
+                        to_mainline?(event.owner, event.name, event.branch) ||
+                            to_pull_request?(event.owner, event.name, event.branch)
+                    elsif event.kind_of? Github::PullRequestEvent
+                        pr = event.pull_request
+                        to_mainline?(pr.base_owner, pr.base_name, pr.base_branch)
+                    else
+                        false
```

Comment:
false is implied (`if .... elsif ... end` is nil if no branch is taken)

Reply:
Yeah.. I don't like returning nil when a boolean is expected. I see this a lot in ruby code though, not sure if it makes a difference at all. I will change to keep it consistent with what you usually do